### PR TITLE
Add clause of replace all to on_conflict

### PR DESCRIPTION
**Pull request by ohta-rh**
HI, Thanks for great Library.
I wanted to be able to set conditions for all columns when specifying the on_conflict clause in insert / insert_all. for example, it was necessary when doing large batch processing with RDB.
therefore, I added a little fix to the library. I would be happy if you could review.

Thanks.

### CODE REVIEW

File: lib/ecto/adapters/postgres/connection.ex [hunk 1]
```diff
@@ -183,6 +183,9 @@ if Code.ensure_loaded?(Postgrex) do
     defp on_conflict({:nothing, _, targets}) do
       "ON CONFLICT " <> conflict_target(targets) <> "DO NOTHING"
     end
+    defp on_conflict({:replace_all, _, targets}) do
+      error!(nil, "The :replace_all option is not supported by PostgreSQL")
```
Comment (josevalim) [inline lib/ecto/adapters/postgres/connection.ex]: We can't support it on PostgreSQL? :S Or you just didn't add support for it?

**Comment by josevalim** [comment]
Thank you! I believe this is a great idea but it would be nice if we could support it across all adapters. :)

File: test/ecto/adapters/mysql_test.exs [hunk 2]
```diff
@@ -515,6 +518,7 @@ defmodule Ecto.Adapters.MySQLTest do
     assert query == ~s{DELETE FROM `prefix`.`schema` WHERE `x` = ? AND `y` = ?}
   end
 
+
```
Comment (ohta-rh) [inline test/ecto/adapters/mysql_test.exs]: thanks for review!! oops,, Fix it!

File: lib/ecto/adapters/postgres/connection.ex (hunk 1 above)
Comment (ohta-rh) [inline lib/ecto/adapters/postgres/connection.ex]: @josevalim Thank you kindly!
I was not using PostgresSQL in my Project, I did not support it, but also understand the syntax of PostgresSQL, Thank you.
I will try the integration test, but I will contact you if I'm in trouble ;)

**Comment by josevalim** [comment]
@ohta-rh because I want to release Ecto 2.1 final this weekend, are you OK with if merge this as is and implement the remaining functionality? If you have already done part of the work locally, feel free to push it and I can continue from there. Thank you! :heart:

**Comment by ohta-rh** [comment]
@josevalim Okey, I'll try to my best, I will send you Pull Request today. Thank you!

**Comment by ohta-rh** [comment]
@josevalim Implementation of PostgresSQL's `:replace_allP and integration test is completed.
Please check my Pull-Request.

Also, what I thought about the integration test.

PostgresSQL is ahead the implementation of `UPSERT`,therefore, control with EXUnit.exclude has become complicated, Although it is a test layer, I'd like to organize it later.

Thank you for various support
(Sorry, broken-English..
)

**Comment by josevalim** [comment]
@ohta-rh your english is excellent. :) I will review it and merge it later today, thank you!

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**